### PR TITLE
refactor: Remove checkout dependencies in address book. (GH-12192)

### LINKED ISSUE #12192: Dependency cleanup -> [storefront] My Account Address Book -> CheckoutDeliveryService, AddressFormModule
Goal: Remove checkout dependencies in the address book feature.

```
projects/storefrontlib/src/cms-components/myaccount/address-book/address-book.component.service.ts:4:3 - error TS2305: Module '"@spartacus/core"' has no exported member 'CheckoutDeliveryService'.
4   CheckoutDeliveryService,
projects/storefrontlib/src/cms-components/myaccount/address-book/address-book.component.ts:6:3 - error TS2305: Module '"@spartacus/core"' has no exported member 'CheckoutDeliveryService'.
6   CheckoutDeliveryService,
projects/storefrontlib/src/cms-components/myaccount/address-book/address-book.module.ts:12:35 - error[0m TS2307: Cannot find module '../../checkout/components/shipping-address/address-form/address-form.module' or its corresponding type declarations.
12 import { AddressFormModule } from '../../checkout/components/shipping-address/address-form/address-form.module';


```

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-12192

### CODE REVIEW

File: feature-libs/checkout/components/components/payment-method/payment-form/payment-form.component.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ import {
 import { Card, ICON_TYPE, ModalRef, ModalService } from '@spartacus/storefront'; // eslint-disable-line
 import { BehaviorSubject, combineLatest, Observable, Subscription } from 'rxjs';
 import { map, switchMap, tap } from 'rxjs/operators';
-import { SuggestedAddressDialogComponent } from '../../shipping-address/address-form/suggested-addresses-dialog/suggested-addresses-dialog.component'; // eslint-disable-line
+import { SuggestedAddressDialogComponent } from '@spartacus/storefront'; // eslint-disable-line
```
Comment (dunqan) [inline feature-libs/checkout/components/components/payment-method/payment-form/payment-form.component.ts]: do we still need this eslint disable?

File: docs/migration/4_0.md [hunk 2]
```diff
@@ -1,3 +1,72 @@
...
+Changes: 
+- PaymentFormComponent does not implement `OnDestroy` anymore
+- method `ngOnDestroy()` removed.
+- Address verification uses new `UserAddressService.verifyAddress` function instead of `CheckoutDeliveryService.verifyAddress`. 
+
+
+## Changes in the classes carried over to checkout lib
+
+To split out the checkout code in the checkout lib, the address verification functionality
+was moved in `UserAddressService` in @spartacus/core.  The address verification related functions in `CheckoutDeliveryService` and ngrx supporting classes are not prresent in the checkout lib.
```
Comment (dunqan) [inline docs/migration/4_0.md]: Some minor misspellings: prresent, Thes, checlout

File: projects/storefrontlib/src/cms-components/myaccount/address-book/address-book.component.service.ts [hunk 3]
```diff
@@ -33,6 +36,31 @@ export class AddressBookComponentService {
 
   updateUserAddress(addressId: string, address: Address) {
     this.userAddressService.updateUserAddress(addressId, address);
-    this.checkoutDeliveryService.clearCheckoutDeliveryDetails();
+    this.eventService.dispatch<UserAddressUpdateEvent>(
```
Comment (dunqan) [inline projects/storefrontlib/src/cms-components/myaccount/address-book/address-book.component.service.ts]: I believe we should dispatch events at the closest place when the actual thing happens, in this case, it should be inside `this.userAddressService.updateUserAddress(addressId, address);`

File: feature-libs/checkout/components/components/payment-method/payment-form/payment-form.component.ts (hunk 1 above)
Comment (plabadie) [reply feature-libs/checkout/components/components/payment-method/payment-form/payment-form.component.ts]: Doesn't look required anymore.. I removed it.

File: docs/migration/4_0.md (hunk 2 above)
Comment (plabadie) [reply docs/migration/4_0.md]: Thanks.  Fixed.

File: projects/storefrontlib/src/cms-components/myaccount/address-book/address-book.component.service.ts (hunk 3 above)
Comment (plabadie) [reply projects/storefrontlib/src/cms-components/myaccount/address-book/address-book.component.service.ts]: Agreed.  Fixed.  I moved all three address change events dispatch to UserAddressService

**Review by dunqan — APPROVED** [review]

**Pull request merged**